src/ui/shared/useScreenViewChange.ts

```diff
@@ -1,36 +1,59 @@
 import { useEffect, useRef } from 'react';
-import { useMutation } from '@tanstack/react-query';
 import { useLocation } from 'react-router-dom';
 import { walletPort } from 'src/ui/shared/channels';
 import { urlContext } from 'src/shared/UrlContext';
 
+async function trackScreenView({
+  event,
+  pathname,
+  previousPathname,
+}: {
+  event: 'screenView' | 'appOpened';
+  pathname: string;
+  previousPathname: string | null;
+}) {
+  const address = await walletPort.request('getCurrentAddress');
+  return walletPort.request(event, {
+    pathname,
+    address,
+    previous: previousPathname,
+    screenSize: `${window.screen.width}x${window.screen.height}`,
+    windowType: urlContext.windowType,
+  });
+}
+
 export function useScreenViewChange() {
   const { pathname } = useLocation();
   const previousPathname = useRef<string | null>(null);
 
-  // TODO:
-  // Refactor to NOT use useMutation, because useMutation makes the hook rerender
-  // for changes to isLoading, isSuccess, etc, and we do not need these updates here.
-  // But you need to make sure that the "previous" param sent to analytics is correct.
-  const { mutate } = useMutation({
-    mutationFn: async (pathname: string) => {
-      const address = await walletPort.request('getCurrentAddress');
-      return walletPort.request('screenView', {
+  const didTrackAppOpened = useRef(false);
+  useEffect(() => {
+    if (!didTrackAppOpened.current) {
+      trackScreenView({
+        event: 'appOpened',
         pathname,
-        address,
-        previous: previousPathname.current,
-        screenSize: `${window.screen.width}x${window.screen.height}`,
-        windowType: urlContext.windowType,
+        previousPathname: previousPathname.current,
+      }).then(() => {
+        previousPathname.current = pathname;
       });
```

Comment:
I think that maybe it doesn't make sense to await the call to update prev path name?
Wdyt?

```
trackScreenView({ ..., previousPathname: previousPathname.current });
previousPathname.current = pathname;
```

Reply:
ah, yess, actually makes sense, no need to wait to update it